src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java

```diff
@@ -37,4 +39,13 @@ public void monitorItemCraft(CraftItemEvent event) {
         island.ifPresent(value -> IridiumSkyblock.getInstance().getIslandManager().incrementMission(value, "CRAFT:" + material.name(), amount));
     }
 
+    @EventHandler(priority = EventPriority.LOW)
+    public void onItemCraft(PrepareItemCraftEvent event) {
+        for (ItemStack itemStack : event.getInventory().getMatrix()) {
+            if (itemStack != null && IridiumSkyblock.getInstance().getIslandManager().getIslandCrystals(itemStack) > 0) {
+                event.getInventory().setResult(new ItemStack(Material.AIR));
```

Comment:
think we should set to null instead, also no need to check if itemStack != null we do that inside the getIslandCrystals method